lib/Backends/NNPI/NNPI.cpp

```diff
@@ -308,6 +330,23 @@ bool NNPIBackend::shouldLower(const Node *N) const {
   case Kinded::Kind::BatchedReduceMeanNodeKind:
   case Kinded::Kind::BatchedReduceMinNodeKind:
     return false;
+  case Kinded::Kind::FusedRowwiseQuantizedSparseLengthsSumNodeKind: {
+    const FusedRowwiseQuantizedSparseLengthsSumNode *SLSN =
+        llvm::cast<FusedRowwiseQuantizedSparseLengthsSumNode>(N);
+    if ((backendOptions_.useIceT || backendOptions_.inferOnDevice) &&
+        (SLSN->getResult().getElementType() == ElemKind::Float16Ty)) {
+      return false; // Don't lower == keep without weights
+    } else {
+      return true;
```

Comment:
So we're lowering (to include weights of 1) only if on ICEREF? I expected there to always be an ICEREF version of anything supported on device, so this seems strange to me.

Reply:
@jfix71 You are correct, ICE-Ref does not support this yet while the HW does.